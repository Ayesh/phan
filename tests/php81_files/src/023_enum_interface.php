<?php

enum A23 {
    case Single;
}

enum B23: int {
    case Single = 1;
}

// This will throw
class Invalid23 implements UnitEnum, Traversable {
    public static function cases(): array {
        return [];
    }
}

function test23(UnitEnum $a, BackedEnum $b) {
    var_dump($a->name, $b->name);
    var_dump($a->value, $b->value);
    $a->name = 'bad';
    $b->name = 'bad';
    $b->value = 'bad';
    $b->extra = 'bad';
    var_dump(clone $a, clone $b);
}
test23(A23::Single, B23::Single);
