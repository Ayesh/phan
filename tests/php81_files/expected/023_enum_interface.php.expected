%s:12 PhanEnumCannotImplement Classlike \Invalid23 cannot implement \UnitEnum in php 8.1+
%s:12 PhanUnreferencedClass Possibly zero references to class \Invalid23
%s:13 PhanPluginUnknownArrayMethodReturnType Method \Invalid23::cases() has a return type of array, but does not specify any key types or value types
%s:20 PhanUndeclaredProperty Reference to undeclared property \UnitEnum->value
%s:21 PhanAccessReadOnlyProperty Cannot modify read-only property \UnitEnum->name defined at internal:0
%s:22 PhanAccessReadOnlyProperty Cannot modify read-only property \BackedEnum->name defined at internal:0
%s:23 PhanAccessReadOnlyProperty Cannot modify read-only property \BackedEnum->value defined at internal:0
%s:24 PhanUndeclaredProperty Reference to undeclared property \BackedEnum->extra
%s:25 PhanTypeInstantiateEnum Saw instantiation of enum \BackedEnum
%s:25 PhanTypeInstantiateEnum Saw instantiation of enum \UnitEnum
